bigbang/archive.py

```diff
@@ -24,89 +33,198 @@ class Archive(object):
     threads = None
     entities = None
 
-    def __init__(self, data, archive_dir=CONFIG.mail_path, mbox=False):
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        mbox: bool = False,
+    ):
         """
         Initialize an Archive object.
 
-        The behavior of the constructor depends on the type
-        of its first argument, data.
-
-        If data is a Pandas DataFrame, it is treated as a representation of
-        email messages with columns for Message-ID, From, Date, In-Reply-To,
-        References, and Body. The created Archive becomes a wrapper around a
-        copy of the input DataFrame.
-
-        If data is a string, then it is interpreted as a path to either a
-        single .mbox file (if the optional argument single_file is True) or
-        else to a directory of .mbox files (also in .mbox format). Note that
-        the file extensions need not be .mbox; frequently they will be .txt.
+        Args:
+            data: representation of email messages with columns for
+                Message-ID, From, Date, In-Reply-To, References, and Body.
+                The created Archive becomes a wrapper around a copy of
+                the input DataFrame.
 
         Upon initialization, the Archive object drops duplicate entries
         and sorts its member variable *data* by Date.
         """
-          
-        if isinstance(data, pd.core.frame.DataFrame):
-            self.data = data.copy()
-        elif isinstance(data, str):
-            self.data = mailman.load_data(data,archive_dir=archive_dir,mbox=mbox)
-        
+        self.data = df
```

Comment:
What happened to the string input case?
The old version copies the data whereas the new version reuses the old dataframe.
So if some other process mutates the dataframe, the new archive is effected.

Why did you make these changes?

Reply:
I have implemented various classmethods for the initialization of the class. This requires the user to now from what data type they want to create the class (e.g. url, string, dataframe). The case for loading the archive from a file is outlined [here](https://github.com/Christovis/bigbang/blob/c47b8ccf0cd9403899269832362a43353c222cc9/bigbang/archive.py#L145). 

It seems to me that this approach is more common now-a-days as outline [here](https://www.programiz.com/python-programming/methods/built-in/classmethod).

We can of course implement a classmethod, from_any(), that initialization the class and assumes now knowledge of what is put in (thus url, string, or dataframe have to distinguished).